porch/apiserver/pkg/e2e/suite.go

```diff
@@ -356,3 +359,187 @@ func createInitialCommit(t *testing.T, repo *gogit.Repository) {
 		t.Fatalf("Failed to set refs/heads/main to commit sha %s", commitHash)
 	}
 }
+
+func inferGitServerImage(porchImage string) string {
+	slash := strings.LastIndex(porchImage, "/")
+	repo := porchImage[:slash+1]
+	image := porchImage[slash+1:]
+	colon := strings.LastIndex(image, ":")
+	tag := image[colon+1:]
+
+	return repo + "git-server:" + tag
+}
+
+func (t *TestSuite) createInClusterGitServer() GitConfig {
+	ctx := context.TODO()
+
+	// Determine git-server image name. Use the same container registry and tag as the Porch server,
+	// replacing base image name with `git-server`. TODO: Make configurable?
+
+	var porch appsv1.Deployment
+	t.GetF(ctx, client.ObjectKey{
+		Namespace: "porch-system",
+		Name:      "porch-server",
+	}, &porch)
+
+	gitImage := inferGitServerImage(porch.Spec.Template.Spec.Containers[0].Image)
+
+	var replicas int32 = 1
+	var selector = strings.ReplaceAll(t.Name(), "/", "_")
+
+	t.CreateF(ctx, &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "git-server",
+			Namespace: t.namespace,
+			Annotations: map[string]string{
+				"kpt.dev/porch-test": t.Name(),
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &replicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"git-server": selector,
+				},
+			},
+			Template: coreapi.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"git-server": selector,
+					},
+				},
+				Spec: coreapi.PodSpec{
+					Containers: []coreapi.Container{
+						{
+							Name:  "git-server",
+							Image: gitImage,
+							Args:  []string{},
+							Ports: []coreapi.ContainerPort{
+								{
+									ContainerPort: 8080,
+									Protocol:      coreapi.ProtocolTCP,
+								},
+							},
+							ImagePullPolicy: coreapi.PullIfNotPresent,
+						},
+					},
+				},
+			},
+		},
+	})
+
+	t.Cleanup(func() {
+		t.DeleteE(ctx, &appsv1.Deployment{
```

Comment:
No need to change, but it might suffice to just delete the namespace

Reply:
When we run against dev instance, namespace controller is not running so deleting namespace wouldn't quite work, but long-term I'd like to get to the point where deleting the namespace would suffice. For now I'll leave it to possibly catch more issues.